azurerm/resource_arm_logic_app_workflow.go

```diff
@@ -0,0 +1,264 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/logic/mgmt/2016-06-01/logic"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+var logicAppResourceName = "azurerm_logic_app"
+
+func resourceArmLogicAppWorkflow() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmLogicAppWorkflowCreate,
+		Read:   resourceArmLogicAppWorkflowRead,
+		Update: resourceArmLogicAppWorkflowUpdate,
+		Delete: resourceArmLogicAppWorkflowDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			// TODO: should Parameters be split out into their own object to allow validation on the different sub-types?
+			"parameters": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+
+			"workflow_schema": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "https://schema.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#",
+			},
+
+			"workflow_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "1.0.0.0",
+			},
+
+			"tags": tagsSchema(),
+
+			"access_endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmLogicAppWorkflowCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).logicWorkflowsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Logic App Workflow creation.")
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	parameters := expandLogicAppWorkflowParameters(d)
+
+	workflowSchema := d.Get("workflow_schema").(string)
+	workflowVersion := d.Get("workflow_version").(string)
+	tags := d.Get("tags").(map[string]interface{})
+
+	properties := logic.Workflow{
+		Location: utils.String(location),
+		WorkflowProperties: &logic.WorkflowProperties{
+			Definition: &map[string]interface{}{
+				"$schema":        workflowSchema,
+				"contentVersion": workflowVersion,
+				"actions":        make(map[string]interface{}, 0),
+				"triggers":       make(map[string]interface{}, 0),
+			},
+			Parameters: parameters,
+		},
+		Tags: expandTags(tags),
+	}
+
+	_, err := client.CreateOrUpdate(ctx, resourceGroup, name, properties)
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error creating Logic App Workflow %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, name)
```

Comment:
I'm guessing this SDK read here is necessary because the SDK create function does not provide the ID -- it feels extraneous to do this and should probably be avoided if possible (e.g. generate the ID from given arguments). The resource read function can catch non-existent IDs.

Reply:
we do this across the provider since the URI used for Create doesn't always match the URI used for Get, unfortunately